app/models/CategorisationAnswers.scala

```diff
@@ -0,0 +1,124 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package models
+
+import cats.data.{EitherNec, NonEmptyChain}
+import cats.implicits.catsSyntaxTuple2Parallel
+import models.AssessmentAnswer.NoExemption
+import models.ott.{CategorisationInfo, CategoryAssessment}
+import org.apache.pekko.Done
+import pages.{AssessmentPage, HasSupplementaryUnitPage, SupplementaryUnitPage}
+import play.api.libs.json.{Json, OFormat}
+import queries.RecordCategorisationsQuery
+
+final case class CategorisationAnswers(
+  assessmentValues: Seq[AssessmentAnswer],
+  supplementaryUnit: Option[Int]
+)
+
+object CategorisationAnswers {
+
+  implicit lazy val format: OFormat[CategorisationAnswers] = Json.format
+
+  private case class CategorisationDetails(index: Int, assessment: CategoryAssessment, answer: AssessmentAnswer)
+
+  def build(userAnswers: UserAnswers, recordId: String): EitherNec[ValidationError, CategorisationAnswers] =
+    (
+      buildAssessmentDetails(userAnswers, recordId),
+      getSupplementaryUnit(userAnswers, recordId)
+    ).parMapN(CategorisationAnswers.apply)
+
+  private def getSupplementaryUnit(userAnswers: UserAnswers, recordId: String) =
+    userAnswers.getOptionalPageValueForOptionalBooleanPage(
+      userAnswers,
+      HasSupplementaryUnitPage(recordId),
+      SupplementaryUnitPage(recordId)
+    )
+
+  private def buildAssessmentDetails(
+    userAnswers: UserAnswers,
+    recordId: String
+  ): Either[NonEmptyChain[ValidationError], Seq[AssessmentAnswer]] =
+    for {
+      recordCategorisations <- userAnswers.getPageValue(RecordCategorisationsQuery)
+      categorisationInfo    <- getCategorisationInfoForThisRecord(recordCategorisations, recordId)
+      answeredAssessments   <- getAssessmentsFromUserAnswers(categorisationInfo, userAnswers, recordId)
+      _                     <- ensureNoExemptionIsOnlyFinalAnswer(answeredAssessments, recordId)
+      _                     <- ensureHaveAnsweredTheRightAmount(answeredAssessments, categorisationInfo.categoryAssessments.size)
+      justTheAnswers         = answeredAssessments.map(_.answer)
+    } yield justTheAnswers
+
+  private def getCategorisationInfoForThisRecord(recordCategorisations: RecordCategorisations, recordId: String) =
+    recordCategorisations.records
+      .get(recordId)
+      .map(Right(_))
+      .getOrElse(Left(NonEmptyChain.one(NoCategorisationDetailsForRecordId(RecordCategorisationsQuery, recordId))))
+
+  private def getAssessmentsFromUserAnswers(
+    categorisationInfo: CategorisationInfo,
+    userAnswers: UserAnswers,
+    recordId: String
+  ): EitherNec[ValidationError, Seq[CategorisationDetails]] = {
+    val answers = categorisationInfo.categoryAssessments.zipWithIndex
+      .map(assessment => (assessment._2, assessment._1, userAnswers.get(AssessmentPage(recordId, assessment._2))))
+      .filter(x => x._3.isDefined)
+      .map(x => CategorisationDetails(x._1, x._2, x._3.get))
```

Comment:
x._3.get - dangerous/need to handle the case where it doesn't exist

what do _1, _2 mean, can these be fields instead of tuples?

Reply:
the .isDefined in the line above is ensuring we only do the .get on ones that exist